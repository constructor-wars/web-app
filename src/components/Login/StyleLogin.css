.form_container {
  display: flex;
  flex-direction: column;
  text-align: left;
  max-width: 50vh;
}

.login_main {
  display: flex;
  flex-direction: column;
  justify-content: center;
  min-height: 884px;
  min-width: 402px;

  background-color: var(--background__color);
}

.login_container {
  align-self: center;
  padding: 50px 0;
}

.label_container {
  width: 136%;
  align-self: center;
}

img{
  background-image: var(--logo);
}

.logo {
  padding-bottom: 50px;
  width: 200px;
  height: 200px;
}

.inputfield {
  background-color: var(--light);
  width: 85%;
  height: 20px;
  margin: 5px 0px;
  border-radius: 5px;
}

.hr_line {
  width: 100%;
  align-self: center;
  /* align-self: center; */
}

.login_button,
.signup_button {
  color: var(--light);
  height: 25px;
  margin: 20px 0px;
  width: 120%;
  align-self: center;
  background-color: var(--dark);
}
