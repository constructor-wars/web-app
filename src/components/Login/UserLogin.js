import React from 'react';

function UserLogin(props) {
  return (
    <form>
      <input
        value={props.value}
        placeholder={props.placeholder}
        onChange={props.onChange}
      />
    </form>
  );
}

export default UserLogin;
