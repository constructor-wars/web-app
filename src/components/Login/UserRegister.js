import React from 'react';

function UserRegister(props) {
  return (
    <form>
      <input className=""placeholder={props.placeholder} />
    </form>
  );
}

export default UserRegister;
