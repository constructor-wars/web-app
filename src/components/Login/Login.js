import React from 'react';
import './StyleLogin.css';
import UserLogin from './UserLogin';
import UserRegister from './UserRegister';
import { EventEmitter } from 'events';
import Logo from '../../images/constructor-labs-badge.png';

export default class Login extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      username: '',
      password: '',
      isLoggedin: false
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(event) {
    event.preventDefault();
    const { name, value } = event.target;
    this.setState({
      [name]: value
    });
  }

  handleSubmit(event) {
    event.preventDefault();
    this.setState({
      isLoggedin: true
    });
    localStorage.setItem('LoggedIn', JSON.stringify(this.state.username));
  }

  render() {
    console.log(this.state);
    return (
      <div className="login_main">
        <div className="login_container">
          <img className="logo" src={Logo} />
          <form className="form_container" onSubmit={this.handleSubmit}>
            <UserLogin
              placeholder="enter your user name"
              value={this.state.username}
              onChange={this.handleChange}
              name="username"
              label="✉️"
            />
            <UserLogin
              value={this.state.password}
              placeholder="enter your password"
              onChange={this.handleChange}
              name="password"
              label="🔐"
            />

            <button className="login_button">Log in</button>
            <a href="#"> Forgot Password?</a>
            <hr className="hr_line" />
            <button className="signup_button">Sign Up</button>
          </form>
        </div>
      </div>
    );
  }
}
