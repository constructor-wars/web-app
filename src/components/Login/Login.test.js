import React from "react";
import Login from "./Login";
import renderer from "react-test-renderer";

describe("Login", () => {
  it("matches the snapshot", () => {
    const tree = renderer.create(<Login />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
