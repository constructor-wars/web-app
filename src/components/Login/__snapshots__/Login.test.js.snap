// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login matches the snapshot 1`] = `
<div
  className="login_main"
>
  <div
    className="login_container"
  >
    <img
      className="logo"
      src="test-file-stub"
    />
    <form
      className="form_container"
      onSubmit={[Function]}
    >
      <label
        className="label_container"
      >
        ‚úâÔ∏è
        <input
          className="inputfield"
          name="username"
          onChange={[Function]}
          placeholder="enter your user name"
          value=""
        />
      </label>
      <label
        className="label_container"
      >
        üîê
        <input
          className="inputfield"
          name="password"
          onChange={[Function]}
          placeholder="enter your password"
          value=""
        />
      </label>
      <button
        className="login_button"
      >
        Log in
      </button>
      <a
        href="#"
      >
         Forgot Password?
      </a>
      <hr
        className="hr_line"
      />
      <button
        className="signup_button"
      >
        Sign Up
      </button>
    </form>
  </div>
</div>
`;
